src/com/itacademy/eshop/shop/ShoppingCart.java

```diff
@@ -24,6 +24,33 @@ public String getTotalPrice() {
         /**
          * Cakculates total price of shopping cart
          */
-        throw new UnsupportedOperationException("getTotalPrice() is not implemented yet");
+        double sum = 0;
+        for (int i = 0; i < products.size(); i++) {
+            sum += products.get(i).getPrice();
+        }
+        return String.valueOf(sum);
+    }
+
+    public void addProductToShoppingCart(Product thing) {
```

Comment:
once again, context. thing is not a very good name. try something like product because your method says that you will add a product to a shopping cart